src/util-file.h

```diff
@@ -88,8 +88,12 @@ typedef struct File_ {
     uint8_t md5[SC_MD5_LEN];
     SCSha1 *sha1_ctx;
     uint8_t sha1[SC_SHA1_LEN];
+    uint32_t sha1_num_bytes;
     SCSha256 *sha256_ctx;
     uint8_t sha256[SC_SHA256_LEN];
+    uint32_t sha256_num_bytes;
```

Comment:
Do we want 3 different numbers for the 3 hashes ? (Cannot it be the same ?)

Reply:
I figured it was safer to have the length "attached" to each hash value.
I definitely can reduce it to only one length for all hashes if you want.  Would require a little extra logic to ensure it's only updated once per hashing cycle, but otherwise straightforward.

Reply:
You can use `DEBUG_VALIDATE_BUG_ON` to ensure some assertions